ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStore.java

```diff
@@ -165,13 +165,10 @@ public void delete(List<MessageInfo> infos) throws StoreException {
       for (MessageInfo msg : infos) {
         cloudDestination.deleteBlob((BlobId) msg.getStoreKey(), msg.getOperationTimeMs(), msg.getLifeVersion(), null);
       }
-    } catch (CloudStorageException e) {
-      if (e.getStatusCode() == HttpStatus.SC_NOT_FOUND) {
-        throw new StoreException(e.getMessage(), StoreErrorCodes.ID_Not_Found);
-      }
-      throw new StoreException(e.getMessage(), StoreErrorCodes.IOError);
     } catch (StoreException e) {
       throw e;
+    } catch (Throwable e) {
```

Comment:
Any reason to not have a SC_NOT_FOUND?

Reply:
When calling these fn that update some metadata or deleting a blob, the blob must exist. This is checked by replication logic before calling these fn. If the blob doesn't exist for some reason, I changed the code to return a null.

Reply:
make sense